lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp

```diff
@@ -602,12 +602,12 @@ class ConvertAtenIndexTensorOp : public OpConversionPattern<AtenIndexTensorOp> {
       for (auto i : llvm::seq(startIndex, startIndex + maxRank)) {
         auto resultDimSize = refinedResultShape[i];
         if (ShapedType::isDynamic(resultDimSize)) {
-          return rewriter.notifyMatchFailure(
-              op, "unimplemented: index tensors must have static shape if "
-                  "there is more than one index tensor");
+          broadcastedIndexShape.push_back(
+              getDimOp(rewriter, loc, indexTensors[i - startIndex], 0));
+        } else {
+          broadcastedIndexShape.push_back(getConstant(
```

Comment:
For this type of case analysis, we should be adding .mlir lit tests demonstrating each case -- this makes it easy to refactor the pattern later.